### PR TITLE
Issue/1070 Making BlocksOnCylindrical projection tests faster

### PR DESCRIPTION
**Pull request by markus-jehl**
This PR is about making the BlocksOnCylindrical projection tests faster to avoid AppVeyor timeouts. Current timings with latest commit are below.

-------- Testing Blocks Geometry --------
symmetry test took: 0.947663 seconds
plane symmetry test took: 1.74959 seconds
voxelOnCartesianGrid test took: 0.009056 seconds
axial projection test took: 7.87713 seconds
map orientation test took: 1.40988 seconds
intersection with cylinder test took: 0.122833 seconds

-------- Testing Parallelproj --------
symmetry test took: 1.02611 seconds
plane symmetry test took: 2.07785 seconds
map orientation test took: 0.629038 seconds
axial projection test took: 2.79315 seconds
projection cpu vs. gpu comparison test took: 8.15366 seconds

### CODE REVIEW

**Comment by markus-jehl** [comment]
@danieldeidda could you take a look as well, in case I massacred one of your tests?

**Comment by KrisThielemans** [comment]
thankd @markus-jehl I guess this PR has a wrong title?

**Comment by markus-jehl** [comment]
It does - I actually just piggy-backed on the old ticket to add parallelproj tests and didn't create a new issue (considering this a follow-up job). I updated the title to make it a bit clearer.

**Review by danieldeidda — COMMENTED** [review]
looks good! just a comment

File: src/recon_test/test_blocks_on_cylindrical_projectors.cxx [hunk 1]
```diff
@@ -103,9 +108,37 @@ BlocksTests::set_blocks_projdata_info(shared_ptr<Scanner> scanner_sptr)
...
-      = std::make_shared<TProjDataInfo>(scanner_sptr, num_axial_pos_per_segment, min_ring_diff_v, max_ring_diff_v,
-                                        scanner_sptr->get_max_num_views(), scanner_sptr->get_max_num_non_arccorrected_bins());
+  auto proj_data_info_blocks_sptr = std::make_shared<TProjDataInfo>(
+      scanner_sptr, num_axial_pos_per_segment, min_ring_diff_v, max_ring_diff_v,
+      scanner_sptr->get_max_num_views() / view_fraction, scanner_sptr->get_max_num_non_arccorrected_bins() / bin_fraction);
+
+  return proj_data_info_blocks_sptr;
+}
+
+/*! Function to generate projection data containing only segment 0. Also allows reducing no. of views and tangential bins. */
```
[the remark this answers is not in the document]
Comment (markus-jehl) [reply src/recon_test/test_blocks_on_cylindrical_projectors.cxx:118]: Actually, this function always creates direct projdata (or more or less direct: segments -1, 0 and 1 are created). I left the other one to create all segments (but also added the view and tangential bin reduction option there). This reduction is just modifying the total amount of views and tangential bins that are created - so my understanding is that it still samples the whole range, but at larger intervals.

[the remark this answers is not in the document]
Comment (danieldeidda) [reply src/recon_test/test_blocks_on_cylindrical_projectors.cxx:118]: OK sorry, yes that makes sense

[the remark this answers is not in the document]
Comment (KrisThielemans) [reply src/recon_test/test_blocks_on_cylindrical_projectors.cxx:118]: ok to resolve this comment then and merge?

**Comment by KrisThielemans** [comment]
@markus-jehl given the convoluted commit history, ok to squash-merge?

File: src/recon_test/test_blocks_on_cylindrical_projectors.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieldeidda) [reply src/recon_test/test_blocks_on_cylindrical_projectors.cxx:118]: yes

**Pull request merged**